examples/pod-security-groups/index.ts

```diff
@@ -0,0 +1,196 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+import * as iam from "./iam";
+import * as k8s from "@pulumi/kubernetes";
+import * as app from "./application";
+
+// IAM roles for the node groups.
+const role = iam.createRole("pod-security-policies");
+
+// Create a new VPC
+const eksVpc = new awsx.ec2.Vpc("pod-security-policies", {
+  enableDnsHostnames: true,
+  cidrBlock: "10.0.0.0/16",
+});
+
+// Create an EKS cluster.
+const cluster = new eks.Cluster("pod-security-policies", {
+  skipDefaultNodeGroup: true,
+  vpcId: eksVpc.vpcId,
+  authenticationMode: eks.AuthenticationMode.Api,
+  // Public subnets will be used for load balancers
+  publicSubnetIds: eksVpc.publicSubnetIds,
+  // Private subnets will be used for cluster nodes
+  privateSubnetIds: eksVpc.privateSubnetIds,
+  vpcCniOptions: {
+    // required for security groups for pods
+    enablePodEni: true,
+    // enables  using liveness or readiness probes with security groups for pods
+    disableTcpEarlyDemux: true,
+    configurationValues: {
+      env: {
+        // all inbound/outbound traffic from pod with security group will be enforced by security group rules
+        POD_SECURITY_GROUP_ENFORCING_MODE: "strict",
+      }
+    }
+  },
+});
+
+// Allows the cluster to manage ENIs, required for security groups for pods
+new aws.iam.RolePolicyAttachment("eks-vpc-cni-policy", {
+  policyArn: "arn:aws:iam::aws:policy/AmazonEKSVPCResourceController",
+  role: cluster.core.clusterIamRole.name,
+});
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+
+const ng = eks.createManagedNodeGroup("pod-security-policies", {
+  scalingConfig: {
+    minSize: 1,
+    maxSize: 2,
+    desiredSize: 1,
+  },
+  // Pod Security Groups require instances that support ENI trunking. See: https://docs.aws.amazon.com/eks/latest/userguide/security-groups-for-pods.html
+  instanceTypes: ["c6i.large", "c7i.large"],
+  cluster: cluster,
+  nodeRole: role,
+});
+
+const kube = cluster.provider;
+
+// Create a SecurityGroup for the nginx application
+const nginxSg = new aws.ec2.SecurityGroup("nginx", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster to the nginx SecurityGroup
+configureClusterAccess("nginx", cluster, nginxSg);
+// Create the nginx application (Deployment, Service, SecurityGroupPolicy)
+const nginx = app.createApplication("nginx", nginxSg.id, kube);
+
+/*
+ * Verify that the SecurityGroupPolicy is working as expected.
+ */
+
+// Create a SecurityGroup for the caller job
+const callerSg = new aws.ec2.SecurityGroup("caller", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster and the caller SecurityGroup
```

Comment:
Consider consistent spacing. My eye not trained on TypeScript style but should this have an empty line before //?

Reply:
Uh, that was by accident. Somehow format on save was turned off